tests/unit/services/worker-test.js

```diff
@@ -149,17 +149,46 @@ test('it resolves promise when event stops', (assert) => {
 	});
 });
 
-test('it unsubscribes from a worker', (assert) => {
+test('it unsubscribes from a worker (passing a callback)', (assert) => {
 	assert.expect(2);
 
 	const callback = () => { };
+	const callback2 = () => { };
+
+	const done = assert.async();
```

Comment:
The `done` function is not necessary if you return a promise in the test case:

```js
return Promise.all([subscriptionPromise, subscriptionPromise2])then(() => {
  service.off('subscription', callback).then(() => {
    assert.equal(service.get('_cache.length'), 1, 'Only the worker associated with the callback has been terminated');
  });
});
```

> Same in the next test case

Reply:
[Updated](https://github.com/BBVAEngineering/ember-web-workers/pull/6/files/ac4255c905f3e1ed49f142d4414329cbeabac178..3f55e7432d3674ee80dbc9278948cca7c3b8ee31) the PR @josex2r